doc/specs/wiki.md

```diff
@@ -0,0 +1,348 @@
+
+* The corresponding comment's document must be deleted from the database.
+
+### 7.0 Tagging articles
+
+**Primary Actor:** A user
+
+**Brief:** Similar to the aliasing system, tagging allows users to search for articles on a broader topic.
+
+**Precondition:** The user has the appropriate user class to edit the article, or has the wiki administration permission enabled.
+
+**Trigger:** The user saves a tag on an article.
+
+**Postcondition:** The tag is saved.
+
+**Basic Flow:**
+
+1. The user types in the tag to add.
+2. The user hits submit, and the tag is saved.
+
+**Functional Requirements:**
+
+* Duplicate or blank tags must be ignored.
+* A new revision must be created when a tag is added.
+
+### 7.1 Removing article tags
+
+**Primary Actor:** A user
+
+**Brief:** Tags must be able to be removed for accuracy.
+
+**Precondition:** The user has the appropriate user class to edit the article and has the wiki management permission enabled, or has the wiki administration permission enabled.
+
+**Trigger:** The user clicks the X button next to a tag.
+
+**Postcondition:** The tag is removed.
+
+**Basic Flow:**
+
+1. The user clicks the X button next to a tag.
+2. The tag is deleted.
+
+**Functional Requirements:**
+
+* A new revision must be created when a tag is removed.
+
+### 8.0 Managing an article's dedicated terms
+
+**Primary Actor:** A user
+
+**Brief:** Dedicated terms allow common searches to resolve to the correct article.
+
+**Precondition:** The user has the appropriate class to edit the article and has the wiki management permission enabled, or has the wiki administration permission enabled.
+
+**Trigger:** The user adds or removes a dedicated term from an article.
+
+**Postcondition:**  N/A
+
+**Basic Flow:**
+
+1. The user adds or removes a dedicated term from an article.
+2. The update is saved.
+
+**Applicable business rules:**
+
+* Dedicated terms must be unique across all articles.
+
+## Data model
+
+**Wiki Articles:**
+* The title of the article.
+* The ID of the user that created the article.
+* An array of unique dedicated terms.
+* The timestamp of the article creation time.
+* The timestamp of the last edit time.
+* A sections subdocument, containing:
+  * The title of the section.
+  * The body of the section.
+  * A sub-section document, containing:
+    * The title of the sub-section.
+    * The body of the sub-section.
+* A revisions subdocument, containing:
+  * The revision title.
+  * The revision body.
+  * The timestamp of the revision date.
+  * The ID of the user that authored the revision.
+* A tags subdocument, containing:
+  * The tag name.
+  * The ID of the user that added the tag.
+  * The timestamp of when the tag was added.
+* A comments subdocument, containing:
+  * The ID of the user that added the comment.
+  * The comment body.
+  * The ID of the user that last edited the comment.
+  * The timestamp of when the comment was created.
+  * The timestamp of when the comment was last edited.
```

Comment:
I think comments should be left out of this. The comments feature should be generic and apply to (mostly) all models.